shared/loadgenerator/loadgenerator.go

```diff
@@ -77,3 +82,30 @@ func (g *GeneratorOptions) RunLoadTest(resolvableDomain bool) (*GeneratorResults
 	r, err := fhttp.RunHTTPTest(g.CreateRunnerOptions(resolvableDomain))
 	return &GeneratorResults{Result: r}, err
 }
+
+// SaveJSON saves the results as Json in the artifacts directory
+func (gr *GeneratorResults) SaveJSON(testName string) error {
+	dir, err := testgrid.GetArtifactsDir()
+	if err != nil {
+		return err
+	}
+
+	outputFile := dir + "/" + testName + jsonExt
+	log.Printf("Storing json output in %s", outputFile)
+	f, err := os.OpenFile(outputFile, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
+	if err != nil {
+		return err
+	}
+
+	defer f.Close()
+	json, err := json.Marshal(gr)
+	if err != nil {
+		return err
+	}
+	_, err = f.Write(json)
```

Comment:
Merge with the line below? (holding so you can act on this if you want)

/lgtm
/hold

Reply:
done